### PR TITLE
[IOS-533]Add auto night mode

### PR DESCRIPTION
**Pull request by Siemian**
### Ticket
[IOS-533](https://netguru.atlassian.net/browse/IOS-533)

### Task Description
Goal of this task was implementing feature for automatic switch to night mode based on sun state in users location.

### CODE REVIEW

File: Inbbbox/Source Files/Helpers/NightModeHoursProvider.swift [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        }
+    }
+    
+    class func currentColorModeBasedOnTime() -> ColorMode? {
+        if let sunrise = NightModeHoursProvider.nextSunrise, let sunset = NightModeHoursProvider.nextSunset {
+            
+            let date = Date()
+            let timeUntilSunset = sunset.timeIntervalSince(date)
+            let timeUntilSunrise = sunrise.timeIntervalSince(date)
+            return timeUntilSunset < timeUntilSunrise ? .dayMode : .nightMode
```
Comment (pikor) [inline Inbbbox/Source Files/Helpers/NightModeHoursProvider.swift]: Hmmm..wait. If time to sunset is lower than time to sunrise, that means sunset is nearer that sunrise, which means: .nightMode should be returned.
`return timeUntilSunset < timeUntilSunrise ? .nightMode : .dayMode`
Don't you think?

File: Inbbbox/Source Files/Providers/ColorModeProvider.swift [hunk 2]
```diff
@@ -9,17 +9,21 @@
 import Foundation
 
 enum ColorMode: String {
-    case DayMode
-    case NightMode
+    case dayMode
```
Comment (pikor) [inline Inbbbox/Source Files/Providers/ColorModeProvider.swift]: 👍🏼

File: Inbbbox/Source Files/Helpers/NightModeHoursProvider.swift [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    static let nextSunset = DefaultsKey<Date?>("nextSunset")
+}
+
+// Provides informations about estimated time of
+// sunrise and sundown in users location
+final class NightModeHoursProvider {
+    
+    private static let secondsInDay = 24 * 60 * 60
+    
+    class var nextSunrise: Date? {
```
Comment (pikor) [inline Inbbbox/Source Files/Helpers/NightModeHoursProvider.swift]: Good idea with saving these values to defaults! 👍🏼

File: Unit Tests/EstimatedLocationProviderSpec.swift [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+import Foundation
+import Dobby
+import Quick
+import Nimble
+import Mockingjay
+
+@testable import Inbbbox
+
+class EstimatedLocationProviderSpec: QuickSpec {
```
Comment (pikor) [inline Unit Tests/EstimatedLocationProviderSpec.swift]: 👍🏼

File: Unit Tests/NightModeHoursProviderSpec.swift [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+import Foundation
+import Dobby
+import Quick
+import Nimble
+import Mockingjay
+
+@testable import Inbbbox
+
+class NightModeHoursProviderSpec: QuickSpec {
```
Comment (pikor) [inline Unit Tests/NightModeHoursProviderSpec.swift]: 👍🏼

File: Inbbbox/Source Files/AppDelegate.swift [hunk 6]
```diff
@@ -48,7 +48,21 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             launchedShortcut = shortcut
             shouldPerformAdditionalDelegateHandling = false
         }
-
+        let nightModeHoursProvider = NightModeHoursProvider()
```
Comment (Siemian) [inline Inbbbox/Source Files/AppDelegate.swift]: i wanted to make it explicit what's going on, that's was my reason why it's here

File: Inbbbox/Source Files/Helpers/NightModeHoursProvider.swift (hunk 1 above)
Comment (Siemian) [inline Inbbbox/Source Files/Helpers/NightModeHoursProvider.swift]: no no ;)
`sunset` and `sunrise` are dates of next sunset and sunrise which means that if we have 2 PM next sunrise will be next day. So `timeUntilSunset < timeUntilSunset` and next switch will be from dayMode to nightMode. This means that we should have dayMode now.

Comment (pikor) [inline Inbbbox/Source Files/Helpers/NightModeHoursProvider.swift]: Ahh, okay :)

**Comment by netguru-hound** [comment]
## [Current coverage](https://codecov.io/gh/netguru/inbbbox-ios/pull/347?src=pr) is 31.66% (diff: 84.40%)


> No coverage report found for **develop** at a5b613f.

> Powered by [Codecov](https://codecov.io?src=pr). Last update [a5b613f...62df50c](https://codecov.io/gh/netguru/inbbbox-ios/compare/a5b613ffb6d1bb0e2b56878044636d545de84771...62df50c3bb6b99230d92a982ae5ad8869b8d3ab3?src=pr)

**Pull request merged**